src/main/java/gregtech/api/worldgen/config/WorldGenRegistry.java

```diff
@@ -162,17 +177,32 @@ private static void extractJarVeinDefinitions(Path worldgenRootPath) throws IOEx
             } else {
                 throw new IllegalStateException("Unable to locate absolute path to worldgen root directory: " + sampleUri);
             }
-            GTLog.logger.info("Attempting extraction of standard worldgen definitions from {} to {}",
-                worldgenJarRootPath, worldgenRootPath);
-            List<Path> jarFiles = Files.walk(worldgenJarRootPath)
-                .filter(jarFile -> Files.isRegularFile(jarFile))
-                .collect(Collectors.toList());
-            for (Path jarFile : jarFiles) {
-                Path worldgenPath = worldgenRootPath.resolve(worldgenJarRootPath.relativize(jarFile).toString());
-                Files.createDirectories(worldgenPath.getParent());
-                Files.copy(jarFile, worldgenPath, StandardCopyOption.REPLACE_EXISTING);
+
+            if(targetPath.compareTo(worldgenRootPath) == 0) {
+                GTLog.logger.info("Attempting extraction of standard worldgen definitions from {} to {}",
+                    worldgenJarRootPath, worldgenRootPath);
+                List<Path> jarFiles = Files.walk(worldgenJarRootPath)
+                    .filter(jarFile -> Files.isRegularFile(jarFile))
+                    .collect(Collectors.toList());
+                for (Path jarFile : jarFiles) {
+                    Path worldgenPath = worldgenRootPath.resolve(worldgenJarRootPath.relativize(jarFile).toString());
+                    Files.createDirectories(worldgenPath.getParent());
+                    Files.copy(jarFile, worldgenPath, StandardCopyOption.REPLACE_EXISTING);
+                }
+                GTLog.logger.info("Extracted {} builtin worldgen definitions into worldgen folder", jarFiles.size());
+            }
+            else if(targetPath.compareTo(dimensionsRootPath) == 0) {
+                GTLog.logger.info("Attempting extraction of standard dimension definitions from {} to {}",
+                    worldgenJarRootPath, dimensionsRootPath);
+
+                Path dimensionFile = worldgenJarRootPath.resolve("dimensions.json");
+
+                Path worldgenPath = dimensionsRootPath.resolve(worldgenJarRootPath.relativize(worldgenJarRootPath).toString());
+                Files.copy(dimensionFile, worldgenPath, StandardCopyOption.REPLACE_EXISTING);
+
+                GTLog.logger.info("Extracted builtin dimension definitions into worldgen folder");
```

Comment:
This function is called if the files do not exist. If the files do not exist, they are created in the config folder as the default files. The vein data extraction is done elsewhere in `reinitializeRegisteredVeins` if the files do exist, for the ore veins.

For the dimensions file, I opted to create a new function that is called from `reinitializeRegisteredVeins`, instead of putting the logic directly in `reinitializeRegisteredVeins`. This function gets passed the json object, which is either the default or the one defined by the user.

Reply:
I tried to extract this somehow but I was not able to get results I would like. So let's leave it as it is.